cl/Makefile.chk

```diff
@@ -15,15 +15,18 @@
 # You should have received a copy of the GNU General Public License
 # along with predator.  If not, see <http://www.gnu.org/licenses/>.
 
-HEADERS ?= *.h *.hh
+C_HEADERS ?= *.h
+CXX_HEADERS ?= *.hh
 
 SINK ?= /dev/null
 
 CXXFLAGS += -pedantic -Wall -Wextra -Wno-variadic-macros -Werror -I../include
 
-.PHONY: check $(HEADERS)
+.PHONY: check $(C_HEADERS) $(CXX_HEADERS)
 
 # check whether each header is self-compilable
-check: $(HEADERS)
-$(HEADERS):
+check: $(C_HEADERS) $(CXX_HEADERS)
+$(C_HEADERS):
+	$(CC) $(CFLAGS) -o $(SINK) -c $@
```

Comment:
Could we check `$(C_HEADERS)` by both `$(CC)` and `$(CXX)`?

Reply:
But I like it as it is. It is more consistent" gcc -> h, g++ -> hh

Reply:
The C header files are meant to be valid C and valid C++ at the same time.  This is difficult to check by single invocation of a compiler.